### ISSUE TITLE
S2583: False positive when using a loop and multiple variables

### ISSUE THREAD

**Issue by jairbubbles** [SonarSource/sonar-csharp#1412]
### Description

False positive on the S2583 rule:

![image](https://user-images.githubusercontent.com/15875066/40229874-e3098252-5a95-11e8-9683-ba70eb752825.png)

### Repro steps

Please provide the steps required to reproduce the problem

1. Analyse that code

```csharp
public static void ParseCommandLine(string commandLine)
{
    var value = new System.Text.StringBuilder(commandLine.Length);
    bool inQuote = false;
    bool startedWithQuote = false;

    foreach (var paramChar in commandLine)
    {
        switch (paramChar)
        {
            case '"':
                inQuote = !inQuote;

                if (value.Length == 0)
                    startedWithQuote = true;
                else
                    value.Append('"');
                break;

            case ' ':
                if (inQuote)
                {
                    value.Append(' ');
                }
                else if (value.Length != 0 || startedWithQuote)
                {
                    value.Clear();
                    startedWithQuote = false;
                }
                break;
        }
    }
}
```

2. Calls it with `ParseCommandLine("\"\" ");`

3. See that we enter the condition that SonarQube expects to be always false

### Related information

* SonarC# 7.1
* Visual Studio Version 15.6.6

**Comment by valhristov** [comment]
Thanks for reporting this problem. We already have a ticket that addresses the same issue: #1187, I'll close this one and keep the reference.

**Comment by jairbubbles** [comment]
Fair enough, just wanted to share the code snippet!
